rust/template/api.rs

```diff
@@ -625,7 +626,7 @@ pub unsafe extern "C" fn ddlog_transaction_commit_dump_changes(
             if let Some(f) = f {
                 for (table_id, table_data) in changes.as_ref().iter() {
                     for (val, weight) in table_data.iter() {
-                        debug_assert!(*weight == 1 || *weight == -1);
+                        assert!(*weight == 1 || *weight == -1);
```

Comment:
Is it still possible for the weight to be 0? If not we should clarify it in the c wrapper in `ddlog.h`

Reply:
Ah I see the C wrapper takes a bool. In general, when are the weights allow to be 0? Any time before `consolidate` is called?

Reply:
This particular code runs after all changes have been accumulated, at which point weights must be 1 or -1 since output relations are distinct (regardless of consolidate). This is why the `commit_dump_changes` callback takes `bool`.